packages/pulseaudio/module-sles-sink.c

```diff
@@ -239,99 +290,81 @@ static void process_rewind(pa_sink *s) {
 
 int pa__init(pa_module*m) {
     struct userdata *u = NULL;
+    pa_modargs *ma = NULL;
+    pa_sink_new_data sink_data;
     pa_sample_spec ss;
     pa_channel_map map;
-    pa_modargs *ma = NULL;
-    pa_sink_new_data data;
     uint32_t latency = 0;
 
     pa_assert(m);
 
     m->userdata = u = pa_xnew0(struct userdata, 1);
+    if (!(ma = pa_modargs_new(m->argument, valid_modargs))) {
+        pa_log("Failed to parse module arguments.");
+        goto fail;
+    }
 
-    u->core = m->core;
+    u->core   = m->core;
     u->module = m;
-    u->rtpoll = pa_rtpoll_new();
 
+    u->rtpoll = pa_rtpoll_new();
     if (pa_thread_mq_init(&u->thread_mq, m->core->mainloop, u->rtpoll) < 0) {
         pa_log("pa_thread_mq_init() failed.");
         goto fail;
     }
-
-    /* The queue linking the AudioTrack thread and our RT thread */
-    u->sles_msgq = pa_asyncmsgq_new(0);
-    if (!u->sles_msgq) {
+    if (!(u->sles_msgq = pa_asyncmsgq_new(0))) {
         pa_log("pa_asyncmsgq_new() failed.");
         goto fail;
     }
-
-    /* The msgq from the AudioTrack RT thread should have an even higher
-     * priority than the normal message queues, to match the guarantee
-     * all other drivers make: supplying the audio device with data is
-     * the top priority -- and as long as that is possible we don't do
-     * anything else */
     u->rtpoll_item = pa_rtpoll_item_new_asyncmsgq_read(u->rtpoll, PA_RTPOLL_EARLY-1, u->sles_msgq);
 
-    if (!(ma = pa_modargs_new(m->argument, valid_modargs))) {
-        pa_log("Failed to parse module arguments.");
-        goto fail;
-    }
+    pa_sink_new_data_init(&sink_data);
+    sink_data.driver = __FILE__;
+    sink_data.module = m;
 
     ss = m->core->default_sample_spec;
-    pa_channel_map_init_stereo(&map);
-
-    switch (ss.format) {
-    case PA_SAMPLE_S16LE:
-    case PA_SAMPLE_S24LE:
-    case PA_SAMPLE_S32LE:
```

Comment:
Additionally I allowed because I found:
https://android.googlesource.com/platform/frameworks/wilhelm/+/refs/heads/master/src/android/android_sles_conversions.h
(No `AUDIO_FORMAT_PCM_8_24_BIT` so no `PA_SAMPLE_S24_32LE`; and I blacklisted u8...intentionally?)
But Big-Endian? I don't think so.